File: UserApiService_C#/AdminApiService/Models/Requests/GetStatisticAboutUserCategoryInCategoryPeriodRequest.cs
using Newtonsoft.Json;

namespace AdminApiService.Models.Requests;

public class GetStatisticAboutUserCategoryInCategoryPeriodRequest : IRequest
{
    [JsonProperty("userCategoryIds")]
    public List<long> userCategoryIds { get; set; }
    [JsonProperty("categoryId")]
    public long categoryId { get; set; }
    [JsonProperty("dateStart")]
    public long dateStart { get; set; }
    [JsonProperty("dateEnd")]
    public long dateEnd { get; set; }

}

File: AvitoStatistcService_Java/src/main/java/ru/espada/avito/statistic/model/request/GetStatisticAboutUserCategoryInCategoryPeriodRequest.java
package ru.espada.avito.statistic.model.request;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
// Запрос на получение предложения об изменении цены на продукт для конкретных групп пользователей в категории в зависимости от времени
public class GetStatisticAboutUserCategoryInCategoryPeriodRequest {

    private ArrayList<Long> userCategoryIds;
    private long categoryId;
    private long dateStart;
    private long dateEnd;

}
